### PR TITLE
[NO QA]: Company card details

### PR DESCRIPTION
**Pull request by narefyev91**
### Details
Company Card details page

### Fixed Issues

$ https://github.com/Expensify/App/issues/47377, https://github.com/Expensify/App/issues/48425, https://github.com/Expensify/App/issues/47379
PROPOSAL:

Preconditions:
We need to add mock information for card in Onyx - i added them inside src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx - just uncomment useEffect
We adding as well some mock data to test export cards account - i used quickbooks online for mock data

### Tests
1. Be sure that Company cards enabled for workspace
2. In the list of company cards - select the first one (should probably connected with mock data)
3. Verify that company card details page is opened and data is displaying in the fields
4. Click on Card name change -> verify that new page is opened -> changed name -> click save -> verify that name is changed on details screen page
5. Click on Export account -> verify that new page with list of available cards is opened -> click on link inside text above the list -> verify that link will redirects you to 'accounting/quickbooks-online/export'
6. Go back to Export account list selection -> choose any other card -> verify that new export account will be shown on card details page
7. Click on update card button -> verify that text is changed to "Updating..." and loading spinner is shown to the right
8. Click in Unassign Card button -> verify that confirmation modal is shown -> click Unassign -> verify that app redirects you to all cards list screen ->verify in Onyx store that card was deleted from that user
9. Refresh the page to put mock data again in store
10. Go to Members page -> find mock member by account id -> click on the row -> verify that user has assigned card
11. Click on assigned card -> verify that card details is opened. All interaction with details should be the same as points above

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Be sure that Company cards enabled for workspace
2. In the list of company cards - select the first one (should probably connected with mock data)
3. Verify that company card details page is opened and data is displaying in the fields
4. Click on Card name change -> verify that new page is opened -> changed name -> click save -> verify that name is changed on details screen page
5. Click on Export account -> verify that new page with list of available cards is opened -> click on link inside text above the list -> verify that link will redirects you to 'accounting/quickbooks-online/export'
6. Go back to Export account list selection -> choose any other card -> verify that new export account will be shown on card details page
7. Click on update card button -> verify that text is changed to "Updating..." and loading spinner is shown to the right
8. Click in Unassign Card button -> verify that confirmation modal is shown -> click Unassign -> verify that app redirects you to all cards list screen ->verify in Onyx store that card was deleted from that user
9. Refresh the page to put mock data again in store
10. Go to Members page -> find mock member by account id -> click on the row -> verify that user has assigned card
11. Click on assigned card -> verify that card details is opened. All interaction with details should be the same as points above

### QA Steps
1. Be sure that Company cards enabled for workspace
2. In the list of company cards - select the first one (should probably connected with mock data)
3. Verify that company card details page is opened and data is displaying in the fields
4. Click on Card name change -> verify that new page is opened -> changed name -> click save -> verify that name is changed on details screen page
5. Click on Export account -> verify that new page with list of available cards is opened -> click on link inside text above the list -> verify that link will redirects you to 'accounting/quickbooks-online/export'
6. Go back to Export account list selection -> choose any other card -> verify that new export account will be shown on card details page
7. Click on update card button -> verify that text is changed to "Updating..." and loading spinner is shown to the right
8. Click in Unassign Card button -> verify that confirmation modal is shown -> click Unassign -> verify that app redirects you to all cards list screen ->verify in Onyx store that card was deleted from that user
9. Refresh the page to put mock data again in store
10. Go to Members page -> find mock member by account id -> click on the row -> verify that user has assigned card
11. Click on assigned card -> verify that card details is opened. All interaction with details should be the same as points above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/ace5bafb-3a4e-4e4f-b6b7-37904b966bbc

https://github.com/user-attachments/assets/0c9fff8e-8263-4969-8f3b-1373b0e7dab9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/fc6d5059-de66-4a11-8958-b6694d7ed791

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/e120f2c7-fcac-4259-ae7c-885f3880dac9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/3cf3c642-8c3e-4d16-b589-028613bc0018

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/3b585e39-689e-49d8-ad8f-827b529004e6

https://github.com/user-attachments/assets/d6debef5-cd26-48cc-81d3-751cf8504429

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/138d0a19-3ec5-4b72-b7e2-a1bcc3e8f695

</details>

### CODE REVIEW

**Comment by dubielzyk-expensify** [comment]
Let us know when you got some screenshots and we'll do a design review 😄

**Comment by narefyev91** [comment]
@dubielzyk-expensify here we go:
Screenshot 2024-09-04 at 09 52 53
Screenshot 2024-09-04 at 10 05 32
Screenshot 2024-09-04 at 09 56 31
Screenshot 2024-09-04 at 09 53 27
Screenshot 2024-09-04 at 09 53 08
Seems like we do not have correct icon for Unassign user - could you please provide it. Thanks!

**Comment by shawnborton** [comment]
Yup, agree with Danny here!

**Comment by dubielzyk-expensify** [comment]
What Danny said and otherwise this is looking good to me 👍

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by allroundexperts** [comment]
@narefyev91 Can you please finish the checklist and make sure that the lint errors are resolved?

**Comment by allroundexperts** [comment]
@narefyev91 On the following screen size, the buttons are stacked too close to each other. The settings button also seems to be clipped.

Screenshot 2024-09-09 at 11 01 21 PM

**Comment by allroundexperts** [comment]
Getting the following console warnings on the card name edit page:

Screenshot 2024-09-09 at 11 08 32 PM

Screenshot 2024-09-09 at 11 08 24 PM

**Comment by allroundexperts** [comment]
@narefyev91 Is this ready for review again?

**Comment by narefyev91** [comment]
> @narefyev91 Is this ready for review again?



> @narefyev91 Is this ready for review again?

Yeah - just pushing latest changes for offline support - which was approved yesterday

**Comment by narefyev91** [comment]
@allroundexperts Hey! Can you please review again? Thanks!

**Comment by allroundexperts** [comment]
Sure. Will review again today!

**Review by allroundexperts — APPROVED** [review]
Looks good!

**Comment by narefyev91** [comment]
Friendly bump on this one @yuwenmemon, Thanks!

**Review by yuwenmemon — APPROVED** [review]

**Comment by yuwenmemon** [comment]
:shipit: Thanks!

**Comment by nkuoch** [comment]
Just saw a [log error](https://github.com/Expensify/Expensify/issues/440574#issuecomment-2451905657) making the call to SetCardExportAccount fail:

`Malformed REQUEST: 'exportAccountDetails', ignoring (should match '{(?:"\w+":.+?)(?:,"\w+":.+?)*}') ~~ value: '[object Object]'`

How are we sending `exportAccountDetails`? JSON string? Example of what we'd expect: `'{"quickbooks_online_export_account_debit_policy_id":"5CF1E5435E6EE1BA","quickbooks_online_export_account_debit":"163"}'`

File: src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                                        color={theme.spinner}
+                                    />
+                                }
+                                description={translate('workspace.moreFeatures.companyCards.lastUpdated')}
+                                title={card?.isLoadingLastUpdated ? translate('workspace.moreFeatures.companyCards.updating') : card?.lastUpdated}
+                                interactive={false}
+                            />
+                            <MenuItemWithTopDescription
+                                description={translate('workspace.moreFeatures.companyCards.transactionStartDate')}
+                                title={card?.startDate ? format(card.startDate, CONST.DATE.FNS_FORMAT_STRING) : ''}
```
Comment (DylanDylann) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:151]: It was causing [this issue](https://github.com/Expensify/App/issues/51915). More explained [here](https://github.com/Expensify/App/issues/51915#issuecomment-2462406800)

File: src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                                label={translate('workspace.moreFeatures.companyCards.cardholder')}
+                                title={displayName}
+                                icon={cardholder?.avatar ?? FallbackAvatar}
+                                iconType={CONST.ICON_TYPE_AVATAR}
+                                description={cardholder?.login}
+                                interactive={false}
+                            />
+                            <MenuItemWithTopDescription
+                                description={translate('workspace.moreFeatures.companyCards.cardNumber')}
+                                title={CardUtils.maskCard(card?.lastFourPAN)}
```
Comment (DylanDylann) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:104]: It caused [this issue](https://github.com/Expensify/App/issues/51912), we need to mask the card name to display 6 first numbers and 4 last numbers

File: src/libs/actions/Policy/Policy.ts [hunk 3]
```diff
@@ -4610,6 +4612,242 @@ function deleteWorkspaceCompanyCardFeed(policyID: string, workspaceAccountID: nu
     API.write(WRITE_COMMANDS.DELETE_COMPANY_CARD_FEED, parameters, onyxData);
 }
 
+function unassignWorkspaceCompanyCard(workspaceAccountID: number, cardID: string, bankName: string) {
```
Comment (DylanDylann) [inline src/libs/actions/Policy/Policy.ts:4615]: We missed to update the optimistic data to apply online pattern B on this action

File: src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    const policy = usePolicy(policyID);
+    const [isUnassignModalVisible, setIsUnassignModalVisible] = useState(false);
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const accountingIntegrations = Object.values(CONST.POLICY.CONNECTIONS.NAME);
+    const connectedIntegration = getConnectedIntegration(policy, accountingIntegrations) ?? connectionSyncProgress?.connectionName;
+
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+    const [allBankCards] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID}_${bank}`);
```
Comment (jayeshmangwani) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx:51]: Coming from this issue https://github.com/Expensify/App/issues/58005, we were using the `bank` value directly from the route `params`, which may cause an error(wrong data) when the page is refreshed. When the page is refreshed, the value becomes encoded. To fix this, we need to`decodeURIComponent` bank before using it, more context: https://github.com/Expensify/App/issues/58005#issuecomment-2710892095

File: src/pages/workspace/companyCards/WorkspaceCompanyCardEditCardNamePage.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_COMPANY_CARDS_ENABLED}
+        >
+            <ScreenWrapper
+                testID={WorkspaceCompanyCardEditCardNamePage.displayName}
+                shouldEnablePickerAvoiding={false}
+                shouldEnableMaxHeight
+            >
+                <HeaderWithBackButton
+                    title={translate('workspace.moreFeatures.companyCards.cardName')}
+                    onBackButtonPress={() => Navigation.goBack(ROUTES.WORKSPACE_COMPANY_CARD_DETAILS.getRoute(policyID, cardID, bank))}
```
Comment (ikevin127) [inline src/pages/workspace/companyCards/WorkspaceCompanyCardEditCardNamePage.tsx:58]: Looks like not passing `{compareParams: false}` here to navigation back led to this issue:
- https://github.com/Expensify/App/issues/59074

**Comment by parasharrajat** [comment]
I am not able to comment on the changes for some reason so posting here. 

https://github.com/Expensify/App/pull/48491/files#diff-456b9101fa053ea20ef1fbc06b69f79d73a3a9d0c1ed335c567e448deefc7208R94

This logic does not compare params while going back which results in new route being set when we open the card details from member profile. It caused https://github.com/Expensify/App/issues/62197

**Pull request merged**